### PR TITLE
Reworked makefiles & docker

### PR DESCRIPTION
**Pull request by nopeslide**
refactored makefiles,
moved pio calls into main makefile,
disabled vscode telemetry

### CODE REVIEW

**Review by ikstream — COMMENTED** [review]
What happened to ``` platformio targets should be pio-flash``` etc. ?

**Comment by nopeslide** [comment]
> What happened to ` platformio targets should be pio-flash` etc. ?

> moved pio calls into main makefile,

see `/Makefile`
I moved them there, because these are not directly docker management related.

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,182 @@
...
+	@make --no-print-directory -C docker pio EXEC="pio run -t clean"
+
+.PHONY: distclean-build
+distclean-build: clean-build
+	rm -rf .pio
+
+### flash targets ###
+
+TARGET += flash
+HELP_flash = flashes project to esp
```
Comment (ikstream) [inline Makefile]: please add info on how to pass dev optionally

File: Makefile [hunk 2]
```diff
@@ -1,41 +1,182 @@
...
+
+.PHONY: distclean-flash
+distclean-flash: clean-flash
 	@true
 
-.PHONY: tests
-tests:
+### monitor targets ###
+
+TARGET += monitor
```
Comment (ikstream) [inline Makefile:109]: please add info on how to pass DEV to target

**Review by ikstream — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (nopeslide) [reply Makefile:109]: target has already `check-dev` as dep, which includes instructions

File: Makefile (hunk 1 above)
Comment (nopeslide) [reply Makefile]: target has already `check-dev` as dep, which includes instructions

Comment (ikstream) [reply Makefile]: But it does not show any infos on dev if ```make help``` is called

Comment (nopeslide) [reply Makefile]:
```
--- VARIABLES ---
DEV
	which device to flash/monitor
	(currently: none)
```
`make help` does name the variable.
an unconfigured device will make `make flash` error with explicit instruction how to use the argument.

Comment (ikstream) [reply Makefile]: That's my point. You have to fail to get information on how to use the make command.

why not provide some information like 
```suggestion
HELP_flash = flashes project to esp. Use DEV=path to provide  path to device or use make dev
```

File: Makefile [hunk 3]
```diff
@@ -1,41 +1,185 @@
...
+HELP_dev = specifies which USB device to connect to
+.PHONY: dev
+dev:
+	echo -n "DEV=" > ./.dev; \
+	(for dev in $$(ls /dev/serial/by-id); do echo "$$(readlink -f /dev/serial/by-id/$$dev) $$dev "; done ) \
+	| dialog \
+		--stdout \
+		--menu "choose which device to flash" 0 0 0 \
+		--file /dev/stdin \
+	>> ./.dev
```
Comment (ikstream) [inline Makefile:147]: Add a clear to clear screen after make dev

```suggestion
	>> ./.dev
	clear
```

**Review by ikstream — CHANGES_REQUESTED** [review]

**Comment by ikstream** [comment]
after setting device with ```make dev ``` i run ```make monitor``` and it fails:
```
make[1]: Nothing to be done for 'check'.
docker run \
        --rm \
        -ti \
         \
        -v pio-home:/home/developer:rw \
        -v /home/ikstream/docs/workspace/esp32-core/docker/..:/mnt:rw \
        -w /mnt \
        --hostname platformio \
         \
        lifesensor/platformio:4.1.0 \
        /bin/sh -c 'sudo chgrp developer /dev/ttyUSB1;          python3                         /home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py                     /mnt/.pio/build/pico32/firmware.elf                      --port /dev/ttyUSB1'
chgrp: cannot access '/dev/ttyUSB1': No such file or directory
--- idf_monitor on /dev/ttyUSB1 115200 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---

Exception in thread Thread-2:
Traceback (most recent call last):
  File "/home/developer/.local/lib/python3.6/site-packages/serial/serialposix.py", line 265, in open
    self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
FileNotFoundError: [Errno 2] No such file or directory: '/dev/ttyUSB1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py", line 133, in _run_outer
    self.run()
  File "/home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py", line 222, in run
    self.serial.open()
  File "/home/developer/.local/lib/python3.6/site-packages/serial/serialposix.py", line 268, in open
    raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg))
serial.serialutil.SerialException: [Errno 2] could not open port /dev/ttyUSB1: [Errno 2] No such file or directory: '/dev/ttyUSB1'
```

The device seems to be not passed to the ```make monitor``` target correctly

File: Makefile (hunk 1 above)
Comment (nopeslide) [reply Makefile]: now i get it.

File: Makefile [hunk 4]
```diff
@@ -1,41 +1,186 @@
...
+.PHONY: distclean-flash
+distclean-flash: clean-flash
 	@true
 
-.PHONY: tests
-tests:
+### monitor targets ###
+
+TARGET += monitor
+HELP_monitor = connects to esp32 via serial
```
Comment (nopeslide) [inline Makefile]:
```suggestion
HELP_monitor = connects to esp32 via serial. Use DEV=path to provide path to device or use make dev
```

File: Makefile [hunk 5]
```diff
@@ -1,41 +1,186 @@
...
+HELP_monitor = connects to esp32 via serial. Use DEV=path to provide path to device or use make dev
+.PHONY: monitor
+monitor: | check-monitor
+	@make --no-print-directory -C docker \
+		pio \
+		EXEC="sudo chgrp developer $(DEV); \
+			python3 \
+				/home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py \
+				/mnt/.pio/build/pico32/firmware.elf \
+				--port $(DEV)"
```
Comment (nopeslide) [inline Makefile]:
```suggestion
				--port $(DEV)" \
		DEV=$(DEV)
```
forgot to passthrough device

**Comment by nopeslide** [comment]
> after setting device with `make dev ` i run `make monitor` and it fails:
> 
> ```
> make[1]: Nothing to be done for 'check'.
> docker run \
>         --rm \
>         -ti \
>          \
>         -v pio-home:/home/developer:rw \
>         -v /home/ikstream/docs/workspace/esp32-core/docker/..:/mnt:rw \
>         -w /mnt \
>         --hostname platformio \
>          \
>         lifesensor/platformio:4.1.0 \
>         /bin/sh -c 'sudo chgrp developer /dev/ttyUSB1;          python3                         /home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py                     /mnt/.pio/build/pico32/firmware.elf                      --port /dev/ttyUSB1'
> chgrp: cannot access '/dev/ttyUSB1': No such file or directory
> --- idf_monitor on /dev/ttyUSB1 115200 ---
> --- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
> 
> Exception in thread Thread-2:
> Traceback (most recent call last):
>   File "/home/developer/.local/lib/python3.6/site-packages/serial/serialposix.py", line 265, in open
>     self.fd = os.open(self.portstr, os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
> FileNotFoundError: [Errno 2] No such file or directory: '/dev/ttyUSB1'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
>     self.run()
>   File "/usr/lib/python3.6/threading.py", line 864, in run
>     self._target(*self._args, **self._kwargs)
>   File "/home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py", line 133, in _run_outer
>     self.run()
>   File "/home/developer/.platformio/packages/framework-espidf/tools/idf_monitor.py", line 222, in run
>     self.serial.open()
>   File "/home/developer/.local/lib/python3.6/site-packages/serial/serialposix.py", line 268, in open
>     raise SerialException(msg.errno, "could not open port {}: {}".format(self._port, msg))
> serial.serialutil.SerialException: [Errno 2] could not open port /dev/ttyUSB1: [Errno 2] No such file or directory: '/dev/ttyUSB1'
> ```
> 
> The device seems to be not passed to the `make monitor` target correctly

should be fixed now

**Review by ikstream — APPROVED** [review]
lgtm

**Pull request merged**